src/tls/boring.rs

```diff
@@ -179,34 +185,44 @@ pub struct TlsGrpcChannel {
 }
 
 /// grpc_connector provides a client TLS channel for gRPC requests.
-pub fn grpc_connector(uri: String) -> Result<TlsGrpcChannel, Error> {
+pub fn grpc_connector(uri: String, root_cert: RootCert) -> Result<TlsGrpcChannel, Error> {
     let mut conn = ssl::SslConnector::builder(ssl::SslMethod::tls_client())?;
 
-    conn.set_verify(ssl::SslVerifyMode::NONE);
-    conn.set_verify_callback(ssl::SslVerifyMode::NONE, |_, x509| {
-        info!("ssl: {:?}", x509.error());
-        // TODO: this MUST verify before upstreaming
-        true
-    });
-
+    let uri = Uri::try_from(uri)?;
+    let is_localhost_call = uri.host() == Some("localhost");
+    conn.set_verify(ssl::SslVerifyMode::PEER);
     conn.set_alpn_protos(Alpn::H2.encode())?;
     conn.set_min_proto_version(Some(ssl::SslVersion::TLS1_2))?;
     conn.set_max_proto_version(Some(ssl::SslVersion::TLS1_3))?;
+    match root_cert {
+        RootCert::File(f) => {
+            conn.set_ca_file(f).map_err(Error::InvalidRootCert)?;
```

Comment:
assuming this errors when the file doesn't exist or is expired, could we add a `#cfg(debug)` guarded block that warns with something to remind devs to `ztunnel-local-bootstrap`.